tests/lib/rules/no-unneeded-ternary.js

```diff
@@ -46,14 +47,126 @@ ruleTester.run("no-unneeded-ternary", rule, {
             }]
         },
         {
+            code: "var a = x >= 2 ? true : false;",
+            output: "var a = x >= 2;",
+            errors: [{
+                message: "Unnecessary use of boolean literals in conditional expression.",
+                type: "ConditionalExpression",
+                line: 1,
+                column: 18
+            }]
+        },
+        {
+            code: "var a = x ? true : false;",
+            output: "var a = !!(x);",
```

Comment:
I don't think we should be doing this type of fix. There's no way we can know that the user will want `!!` as the fix for this and you're introducing parentheses that aren't needed. To me, this these are two steps beyond what we can reasonably infer as the correct fix for this.